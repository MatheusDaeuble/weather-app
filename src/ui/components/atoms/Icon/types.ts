import { Color } from 'ui/theme/colors';
import { ViewStyle } from 'react-native';

export type IconName =
  | 'cloud-outline'
  | 'cloudy-night-outline'
  | 'cloudy'
  | 'sync'
  | 'x-circle-outlined'
  | 'check-circle'
  | 'moon-outline'
  | 'partly-sunny-outline'
  | 'rainy-outline'
  | 'snow-outline'
  | 'sunny-outline'
  | 'thunderstorm-outline'
  | 'wind'
  | 'chevron-down'
  | 'chevron-up';

export type IconColor = Color;

export type IconProps = {
  name: IconName;
  disabled?: boolean;
  color?: IconColor;
  size?: number;
  style?: ViewStyle;
  onPress?: () => void;
};
