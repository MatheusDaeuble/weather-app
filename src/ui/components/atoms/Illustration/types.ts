import { SvgProps } from 'react-native-svg';

export type IllustrationName =
  | 'cloud'
  | 'error'
  | 'cloud-moon'
  | 'clouds'
  | 'cloud-sun'
  | 'location'
  | 'mist'
  | 'moon'
  | 'rain'
  | 'rain-moon'
  | 'rain-sun'
  | 'snow'
  | 'storm'
  | 'sun';

export interface IllustrationProps extends SvgProps {
  name: IllustrationName;
}
