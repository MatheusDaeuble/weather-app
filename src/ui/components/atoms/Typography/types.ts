import { TextProps, TextStyle } from 'react-native';
import { Color } from 'ui/theme/colors';

export type TypographyVariant =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'h5'
  | 'body1'
  | 'body2';

export type TypographyProps = {
  variant?: TypographyVariant;
  color?: Color;
  align?: TextStyle['textAlign'];
  children: React.ReactNode;
  paragraph?: boolean;
  onPress?: () => void;
} & TextProps;
