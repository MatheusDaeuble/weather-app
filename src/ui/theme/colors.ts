export enum Colors {
  // PRIMARY
  PRIMARY = '#2E89DF',

  // SECONDARY
  SECONDARY = '#809FFF',

  WHITE = '#FFFFFF',
  BLACK = '#000000',

  // FEEDBACK COLORS
  SUCCESS = '#0E934C',
  WARNING = '#FF8B2C',
  DANGER = '#EB5757',

  // GRAY SCALE
  GRAY_DARK = '#4B4F58',
  GRAY = '#949494',
  GRAY_LIGHT = '#CECECE',
}

export type Color =
  | 'primary'
  | 'secondary'
  | 'white'
  | 'black'
  | 'success'
  | 'warning'
  | 'danger'
  | 'gray-dark'
  | 'gray'
  | 'gray-light';

export const colors: { [key in Color]: string } = {
  'primary': Colors.PRIMARY,
  'secondary': Colors.SECONDARY,
  'white': Colors.WHITE,
  'black': Colors.BLACK,
  'success': Colors.SUCCESS,
  'danger': Colors.DANGER,
  'warning': Colors.WARNING,
  'gray-dark': Colors.GRAY_DARK,
  'gray': Colors.GRAY,
  'gray-light': Colors.GRAY_LIGHT,
};

export const defaultTheme = {
  colors,
};
