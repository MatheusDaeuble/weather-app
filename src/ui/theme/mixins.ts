import { Dimensions, PixelRatio } from 'react-native';

const { width: WINDOW_WIDTH, height: WINDOW_HEIGHT } = Dimensions.get('window');
const guidelineBaseWidth = 375;
const guidelineBaseHeight = 700;

export const scaleSize = (size: number) =>
  Number(((WINDOW_WIDTH / guidelineBaseWidth) * size).toFixed(0));

export const scaleHeight = (size: number) =>
  Number(((WINDOW_HEIGHT / guidelineBaseHeight) * size).toFixed(0));

export const scaleFont = (size: number) =>
  Number((size * PixelRatio.getFontScale()).toFixed(0));
