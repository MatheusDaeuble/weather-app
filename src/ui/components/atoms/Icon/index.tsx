import React from 'react';
import { IconProps } from './types';
import { Feather, Ionicons } from '@expo/vector-icons';
import { colors } from 'ui/theme/colors';
import { TouchableOpacity } from 'react-native';
import { scaleSize } from 'ui/theme/mixins';

const getIcon = ({
  ...props
}: Omit<IconProps, 'color' | 'onPress'> & { color: string }) => {
  switch (props.name) {
    case 'x-circle-outlined':
      return <Feather {...props} name='x-circle' />;
    case 'check-circle':
      return <Feather {...props} name='check-circle' />;
    case 'sync':
      return <Ionicons {...props} name='sync' />;
    case 'cloud-outline':
      return <Ionicons {...props} name='cloud-outline' />;
    case 'cloudy-night-outline':
      return <Ionicons {...props} name='cloudy-night-outline' />;
    case 'cloudy':
      return <Ionicons {...props} name='cloudy' />;
    case 'moon-outline':
      return <Ionicons {...props} name='moon-outline' />;
    case 'partly-sunny-outline':
      return <Ionicons {...props} name='partly-sunny-outline' />;
    case 'rainy-outline':
      return <Ionicons {...props} name='rainy-outline' />;
    case 'snow-outline':
      return <Ionicons {...props} name='snow-outline' />;
    case 'sunny-outline':
      return <Ionicons {...props} name='sunny-outline' />;
    case 'thunderstorm-outline':
      return <Ionicons {...props} name='thunderstorm-outline' />;
    case 'wind':
      return <Feather {...props} name='wind' />;
    case 'chevron-down':
      return <Ionicons {...props} name='chevron-down' />;
    case 'chevron-up':
      return <Ionicons {...props} name='chevron-up' />;
  }
};

export const iconNames: Array<IconProps['name']> = [
  'cloud-outline',
  'cloudy-night-outline',
  'cloudy',
  'sync',
  'x-circle-outlined',
  'check-circle',
  'moon-outline',
  'partly-sunny-outline',
  'rainy-outline',
  'snow-outline',
  'sunny-outline',
  'thunderstorm-outline',
  'wind',
  'chevron-down',
  'chevron-up',
];

const Icon = ({
  color = 'gray',
  size = 24,
  onPress,
  style,
  disabled,
  ...props
}: IconProps) => {
  const iconSize = scaleSize(size);

  const IconToRender = getIcon({
    ...props,
    size: iconSize,
    color: colors[color],
  });

  return (
    <TouchableOpacity
      style={[style, { width: iconSize }]}
      onPress={onPress}
      disabled={!onPress || disabled}
    >
      {IconToRender}
    </TouchableOpacity>
  );
};

export default Icon;
