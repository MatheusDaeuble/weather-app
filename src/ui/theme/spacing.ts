import { scaleSize } from './mixins';

export const SCALE_28 = scaleSize(28);
export const SCALE_24 = scaleSize(24);
export const SCALE_20 = scaleSize(20);
export const SCALE_16 = scaleSize(16);
export const SCALE_12 = scaleSize(12);
export const SCALE_8 = scaleSize(8);
export const SCALE_4 = scaleSize(4);
