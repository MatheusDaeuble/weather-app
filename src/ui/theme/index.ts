import * as Colors from './colors';
import * as Spacing from './spacing';
import * as TypographyAttributes from './typography';
import * as Mixins from './mixins';

export { TypographyAttributes, Spacing, Colors, Mixins };
