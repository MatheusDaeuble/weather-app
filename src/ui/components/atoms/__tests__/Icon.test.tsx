import React from 'react';
import renderer from 'react-test-renderer';
import { Color, colors } from 'ui/theme/colors';
import Icon, { iconNames } from '../Icon';
import { IconName } from '../Icon/types';

describe('Icon Atom', () => {
  describe('Render Name', () => {
    iconNames.forEach((name: IconName) => {
      it(`Render name: ${name}`, () => {
        const component = renderer.create(<Icon name={name} />).toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('Render size', () => {
    const sizes = [...Array(10).keys()];
    sizes.forEach(size => {
      it(`Render size: ${size}`, () => {
        const component = renderer
          .create(<Icon size={16 + size * 4} name={'cloudy'} />)
          .toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });

  describe('Render color', () => {
    Object.keys(colors).forEach((color: Color) => {
      it(`Render color: ${color}`, () => {
        const component = renderer
          .create(<Icon name={'cloudy'} color={color} />)
          .toJSON();
        expect(component).toMatchSnapshot();
      });
    });
  });
});
