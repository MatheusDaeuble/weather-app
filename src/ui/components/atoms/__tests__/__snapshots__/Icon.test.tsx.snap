// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Atom Render Name Render name: check-circle 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: chevron-down 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: chevron-up 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: cloud-outline 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: cloudy 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: cloudy-night-outline 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: moon-outline 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: partly-sunny-outline 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: rainy-outline 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: snow-outline 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: sunny-outline 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: sync 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: thunderstorm-outline 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: wind 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render Name Render name: x-circle-outlined 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: black 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: danger 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: gray 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: gray-dark 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: gray-light 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: primary 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: secondary 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: success 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: warning 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render color Render color: white 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 0 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 32,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 1 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 40,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 2 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 48,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 3 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 56,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 4 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 64,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 5 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 72,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 6 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 80,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 7 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 88,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 8 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 96,
    }
  }
>
  <Text />
</View>
`;

exports[`Icon Atom Render size Render size: 9 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": 104,
    }
  }
>
  <Text />
</View>
`;
